middleware/validate.js

```diff
@@ -0,0 +1,168 @@
+module.exports = {
+  validator: function (req, res, next) {
+
+    /************************
+     *  ARTICLES
+    ************************/
+    if (req.originalUrl.slice(0, 9) === '/articles') {
+      const title = req.params.title;
+      const articleDB = require('../db/articles.js');
+      const database = articleDB.getArticles().articles;
+
+      // GET '/:title'
+      if (req.method === 'GET' && req.url === `/${title}`) {
+        const params = req.params;
+        const articleIndex = articleDB.findArticle(params.title);
+
+        if (typeof articleIndex !== 'number') {
+          res.status(500);
+          return res.json({ "Error": "Couldn't find that product in our database." });
+        }
+      }
+
+      // GET '/:title/edit'
+      if (req.method === 'GET' && req.url === `/${title}/edit`) {
+        let params = req.params;
+        let articleIndex = articleDB.findArticle(params.title);
+        let article = database[articleIndex];
+
+        if (typeof articleIndex !== 'number') {
+          return res.render('./articles/new', { message: "That article is not in our database, please add it as a new article" });
+        }
+      }
+
+      // POST '/articles'
+      if (req.method === 'POST' && req.url === '/articles') {
+        let body = req.body;
+
+        if (Object.keys(body).length < 3) {
+          return res.render('./products/new', { message: "Please fill out all fields with approppriate information" });
+        }
+
+        for (var key in body) {
+          if (body[key].length < 2) {
+            return res.render('./products/new', { message: "Please fill out all fields with approppriate information" });
+          }
+
+          if (!isNaN(parseInt(body[key]))) {
+            return res.render('./products/new', { message: "Please fill out all fields with approppriate information" });
+          }
+        }
+      }
+
+      // PUT '/:title'
+      if (req.method === "PUT" && req.url === `/${title}`) {
+        let body = req.body;
+        const params = req.params
+        const articleIndex = articleDB.findArticle(params.title);
+
+        if (typeof articleIndex !== 'number') {
+          let data = { message: "That article is not in our database, please add it as a new article" };
+
+          return res.render('./articles/new', data);
+        }
+
+        for (var key in body) {
+          if (body[key].length < 2 || !isNaN(parseInt(body[key]))) {
```

Comment:
Not sure why you're trying to parseInt here. Are you trying to make sure it's not a number? I think it would be fine to just make sure the string have a length.